### PR TITLE
feat(contracts): use create3 for avs deployments

### PR DESCRIPTION
**Pull request by kevinhalliday**
Use create3 for avs deployments

task: https://app.asana.com/0/1206208509925075/1206648863973026

### CODE REVIEW

**Comment by kevinhalliday** [comment]
I haven't tested deployment scripts yet. I'll do that against anvil tomorrow

**Review by Lazar955 — COMMENTED** [review]
👀

File: contracts/script/avs/DeployGoerliAVS.s.sol [hunk 1]
```diff
@@ -22,52 +18,77 @@ import { Script } from "forge-std/Script.sol";
  *      deploy script.
  */
 contract DeployGoerliAVS is Script {
-    uint96 public constant MIN_OPERATOR_STAKE = 1 ether;
-    uint32 public constant MAX_OPERATOR_COUNT = 10;
+    uint64 internal _omniChainId = 165;
+    address internal _create3Factory = address(uint160(uint256(keccak256("TODO"))));
```
Comment (pavelnikolov) [inline contracts/script/avs/DeployGoerliAVS.s.sol:22]: Why do we have `uint160(uint256(...))`? Can't we use only one conversion?

**Review by pavelnikolov — APPROVED** [review]

File: contracts/script/avs/DeployGoerliAVS.s.sol (hunk 1 above)
Comment (kevinhalliday) [reply contracts/script/avs/DeployGoerliAVS.s.sol:22]: sadly no

File: contracts/script/avs/DeployGoerliAVS.s.sol [hunk 2]
```diff
@@ -22,52 +18,77 @@ import { Script } from "forge-std/Script.sol";
...
+        uint256 deployerKey = vm.envUint("DEPLOYER_KEY");
+        address factory = vm.envOr("CREATE3_FACTORY_ADDRESS", _create3Factory);
+        address proxyAdmin = vm.envOr("PROXY_ADMIN", _proxyAdmin);
+        address owner = vm.envOr("AVS_OWNER", _owner);
+        address portal = vm.envOr("PORTAL_ADDRESS", _portal);
+        address ethStakeInbox = vm.envOr("ETH_STAKE_INBOX", _ethStakeInbox);
+        bytes32 salt = vm.envOr("CREATE3_SALT", _create3Salt);
+
+        vm.startBroadcast(deployerKey);
+        deploy(factory, salt, owner, proxyAdmin, portal, omniChainId, ethStakeInbox);
```
Comment (ttarsi) [inline contracts/script/avs/DeployGoerliAVS.s.sol:41]: so we''ll need to have deployed a proxy admin first?

**Review by ttarsi — APPROVED** [review]

File: contracts/script/avs/DeployGoerliAVS.s.sol (hunk 2 above)
Comment (kevinhalliday) [reply contracts/script/avs/DeployGoerliAVS.s.sol:41]: Yes adding a script for that in separate Pr

**Pull request merged**